libraries/stdlib/stdlib_ng.mtlx

```diff
@@ -1160,6 +1160,330 @@
     <output name="out" type="float" nodename="N_range" />
   </nodegraph>
 
+  <!--
+    Node: <randomfloat>
+    Produces a randomized float, based on an 'input' signal and 'seed' value.
+  -->
+  <nodegraph name="NG_randomfloat_float" nodedef="ND_randomfloat_float">
+    <multiply name="N_offset_seed" type="float">
+      <input name="in1" type="float" interfacename="seed" />
+      <input name="in2" type="float" value="43758.5" />
+    </multiply>
+    <subtract name="N_subtract1" type="float">
+      <input name="in1" type="float" interfacename="in" />
+      <input name="in2" type="float" nodename="N_offset_seed" />
+    </subtract>
+    <combine2 name="N_combine2" type="vector2">
+      <input name="in1" type="float" nodename="N_subtract1" />
+      <input name="in2" type="float" nodename="N_offset_seed" />
+    </combine2>
+    <cellnoise2d name="N_cellnoise_1" type="float">
+      <input name="texcoord" type="vector2" nodename="N_combine2" />
+    </cellnoise2d>
+    <range name="N_remapRange" type="float">
+      <input name="in" type="float" nodename="N_cellnoise_1" />
+      <input name="outlow" type="float" interfacename="min" />
+      <input name="outhigh" type="float" interfacename="max" />
+      <input name="doclamp" type="boolean" value="true" />
+    </range>
+    <output name="out" type="float" nodename="N_remapRange" />
+  </nodegraph>
+  <nodegraph name="NG_randomfloat_integer" nodedef="ND_randomfloat_integer">
+    <convert name="N_convert1" type="float">
+      <input name="in" type="integer" interfacename="in" />
+    </convert>
+    <randomfloat name="N_randomfloat1" type="float">
+      <input name="in" type="float" nodename="N_convert1" />
+      <input name="min" type="float" interfacename="min" />
+      <input name="max" type="float" interfacename="max" />
+      <input name="seed" type="float" interfacename="seed" />
+    </randomfloat>
+    <output name="out" type="float" nodename="N_randomfloat1" />
+  </nodegraph>
+
+  <!--
+    Node: <randomvector>
+    Produces a randomized vector2, vector3, or vector4, based on an 'input' signal and 'seed' value.
+  -->
+  <nodegraph name="NG_randomvector_float_vector2" nodedef="ND_randomvector_float_vector2">
```

Comment:
I could be wrong, but I think that it might be simpler and clearer to implement these `randomvector` nodes in terms of `randomfloat`, rather than calling the much more complex `vector4` implementation of `randomvector` under the hood.

My sense is that calling `randomfloat` twice for `vector2`, three times for `vector3`, and so forth would be both more efficient and easier to understand, and wouldn't require any complex `swizzle` operations.

What are your thoughts?

Reply:
That makes sense - I'm not sure why I didn't think of that in the first place!